### PR TITLE
Add payload size metric

### PR DESCRIPTION
**Pull request by lmsilva-wls**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -76,7 +88,7 @@ var (
 			Subsystem: "client",
 			Name:      "response_time_ms",
 			Help:      "the response time in ms of calls to server",
-			Buckets:   []float64{1, 3, 5, 10, 25, 50, 100, 150, 200, 250, 300},
+			Buckets:   defaultLatencyBuckets(),
```
Comment (mahtues) [inline metrics/metrics.go]: 👍

File: app/app.go [hunk 2]
```diff
@@ -163,17 +164,27 @@ func (a *App) metricsReporterInterceptor(
...
-	
+
 	topic := events[0].Topic
 	l := a.log.
 		WithField("route", info.FullMethod).
 		WithField("topic", topic)
 
+	// using JSON to calculate the payload size introduces a small, invariant/constant error related when compared to
+	// the gRPC payload, but should be enough for observability
+	marshalledEvents, _ := json.Marshal(events)
```
Comment (mahtues) [inline app/app.go]: `protobuf` compacts field names. Serializing it into JSON will add field names. The JSON size will be greater than the wire size.

Comment (mahtues) [reply app/app.go]: image

image

File: app/app.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ package app
 
 import (
 	"context"
+	"encoding/json"
```
Comment (mahtues) [inline app/app.go:27]:
```suggestion
	"github.com/golang/protobuf/proto"
```

File: app/app.go [hunk 4]
```diff
@@ -163,17 +164,27 @@ func (a *App) metricsReporterInterceptor(
...
+
 	topic := events[0].Topic
 	l := a.log.
 		WithField("route", info.FullMethod).
 		WithField("topic", topic)
 
+	// using JSON to calculate the payload size introduces a small, invariant/constant error related when compared to
+	// the gRPC payload, but should be enough for observability
+	marshalledEvents, _ := json.Marshal(events)
+	payloadSize := len(marshalledEvents)
```
Comment (mahtues) [inline app/app.go:182]:
```suggestion
	payloadSize := 0
	for _, e := range events {
		payloadSize += proto.Size(e)
	}
```

File: app/app.go (hunk 2 above)
Comment (mahtues) [reply app/app.go]: on second tought, calculate the payload inside the type switch.
image

Comment (lmsilva-wls) [reply app/app.go]: Nice catch! I've updated it

**Review by mahtues — APPROVED** [review]

File: app/app.go (hunk 2 above)
Comment (mahtues) [reply app/app.go]: 👍

**Pull request merged**